File: demo/flutter/flutternew/lib/domain/dto/resp/oss_batch_upload_response.dart
import 'package:json_annotation/json_annotation.dart';

import '../user_id_wire_json.dart';
import 'oss_upload_item_result.dart';

part 'oss_batch_upload_response.g.dart';

@JsonSerializable()
class OssBatchUploadResponse {
  @JsonKey(fromJson: userIdFromWireJson)
  final int? userId;
  final String? bucketName;
  final int? successCount;
  final int? failCount;
  final List<OssUploadItemResult>? items;

  const OssBatchUploadResponse({
    this.userId,
    this.bucketName,
    this.successCount,
    this.failCount,
    this.items,
  });

  factory OssBatchUploadResponse.fromJson(Map<String, dynamic> json) =>
      _$OssBatchUploadResponseFromJson(json);

  Map<String, dynamic> toJson() => _$OssBatchUploadResponseToJson(this);
}

File: demo/app/app/src/main/java/com/vectordemo/domain/dto/http/response/OssBatchUploadResponse.kt
package com.vectordemo.domain.dto.http.response

import com.google.gson.annotations.JsonAdapter
import com.vectordemo.utils.json.WireUserIdJsonDeserializer

data class OssBatchUploadResponse(
    @JsonAdapter(WireUserIdJsonDeserializer::class)
    val userId: String? = null,
    val bucketName: String? = null,
    val successCount: Int? = null,
    val failCount: Int? = null,
    val items: List<OssUploadItemResult>? = null
)
